src/v/cluster/health_monitor_types.h

```diff
@@ -155,6 +155,8 @@ struct partition_status
 };
 
 using partition_statuses_t = chunked_vector<partition_status>;
+using partition_statuses_map_t
+  = chunked_hash_map<model::partition_id, partition_status>;
```

Comment:
why do we need a map? We can do ID based lookups in `chunked_vector` or the `contiguous_range_map` that you made. Both would be more efficent right?